modules/models/tests/behavior_motion_primitive_test.cc

```diff
@@ -101,41 +123,79 @@ TEST(behavior_motion_primitives_plan, behavior_test) {
+
+  auto world1 = make_test_observed_world(0, 0.0, 5.0, 0.0);
+  std::const_pointer_cast<Agent>(world1.GetEgoAgent())
+      ->SetDynamicModel(dynamics);
+  AdjacentLaneCorridors corridors = GetCorridors(world1);
+  EXPECT_TRUE(primitive.IsPreConditionSatisfied(world1, corridors));
+  //  auto traj = primitive.Plan(0.5, world1);
+
+  PrimitiveConstAccStayLane dec_primitive(params, -5.0);
+  EXPECT_TRUE(dec_primitive.IsPreConditionSatisfied(world1, corridors));
+  auto world2 = make_test_observed_world(0, 0.0, 0.0, 0.0);
+  std::const_pointer_cast<Agent>(world2.GetEgoAgent())
+      ->SetDynamicModel(dynamics);
+  AdjacentLaneCorridors corridors2 = GetCorridors(world2);
+  EXPECT_FALSE(dec_primitive.IsPreConditionSatisfied(world2, corridors2));
+
+  PrimitiveConstAccStayLane acc_primitive(params, 4.0);
+  EXPECT_TRUE(acc_primitive.IsPreConditionSatisfied(world1, corridors));
+  //  auto world3 = make_test_observed_world(0, 0.0, 50.0, 0.0);
+  //  std::const_pointer_cast<Agent>(world3.GetEgoAgent())->SetDynamicModel(dynamics);
+  //  AdjacentLaneCorridors corridors3 = GetCorridors(world3);
+  //  EXPECT_FALSE(acc_primitive.IsPreConditionSatisfied(world3, corridors));
 }
 
 TEST(primitive_change_left, behavior_test) {
   using modules::models::behavior::primitives::PrimitiveConstAccChangeToLeft;
   auto params = std::make_shared<DefaultParams>();
-  DynamicModelPtr dynamics(new SingleTrackModel(params));
-  PrimitiveConstAccChangeToLeft primitive(params, dynamics, 0.1);
+  PrimitiveConstAccChangeToLeft primitive(params);
+  auto world = MakeTestWorldHighway();
+  auto observed_worlds = world->Observe({1, 4});
+  auto corridors0 = GetCorridors(observed_worlds[0]);
+  EXPECT_FALSE(
+      primitive.IsPreConditionSatisfied(observed_worlds[0], corridors0));
+  auto corridors1 = GetCorridors(observed_worlds[1]);
+  EXPECT_TRUE(
+      primitive.IsPreConditionSatisfied(observed_worlds[1], corridors1));
+  // auto traj = primitive.Plan(0.5, world1);
+}
 
-  State init_state(static_cast<int>(StateDefinition::MIN_STATE_SIZE));
-  init_state << 0.0, 0.0, 0.0, 0.0, 5.0;
-  auto world1 = std::make_shared<DummyObservedWorld>(init_state, params);
-  // EXPECT_FALSE(primitive.IsPreConditionSatisfied(world1));
+TEST(primitive_change_right, behavior_test) {
+  using modules::models::behavior::primitives::PrimitiveConstAccChangeToRight;
+  auto params = std::make_shared<DefaultParams>();
+  PrimitiveConstAccChangeToRight primitive(params);
+  auto world = MakeTestWorldHighway();
+  for (const auto& a : world->GetAgents()) {
+    LOG(WARNING) << a.first << ", ";
+  }
+  auto observed_worlds = world->Observe({1, 4});
+  auto corridors0 = GetCorridors(observed_worlds[0]);
+  EXPECT_TRUE(
+      primitive.IsPreConditionSatisfied(observed_worlds[0], corridors0));
+  auto corridors1 = GetCorridors(observed_worlds[1]);
+  EXPECT_FALSE(
+      primitive.IsPreConditionSatisfied(observed_worlds[1], corridors1));
   // auto traj = primitive.Plan(0.5, world1);
```

Comment:
What is tested here is the precondition. Can we test for the action as well? Otherwise, rename the test function